src/GraphQL/Types/Collections/IGraphTypeMappingResolver.cs

```diff
@@ -0,0 +1,16 @@
+namespace GraphQL.Types;
+
+/// <summary>
+/// Provides a mapping from CLR types to graph types.
+/// </summary>
+public interface IGraphTypeMappingResolver
```

Comment:
"Resolver" in name may sound a bit confusing.

Reply:
I'm open to suggestions.  I added Resolver because it doesn't represent a mapping (with a concrete set of types) so much as a delegate / factory / resolver.

Reply:
Resolver -> Provider

The only reason from me - do not mix this API with field resolvers.

Reply:
I agree.

Reply:
done.  I hope I didn't miss anything.